src/lando/main/models.py

```diff
@@ -1 +1,181 @@
-# Create your models here.
+from __future__ import annotations
+
+import datetime
+import logging
+import os
+from typing import (
+    Any,
+    Iterable,
+    Optional,
+)
+
+from django.db import models
+from django.db.models import QuerySet
+from django.utils.translation import gettext_lazy
+
+from lando.utils import build_patch_for_revision
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_GRACE_SECONDS = int(os.environ.get("DEFAULT_GRACE_SECONDS", 60 * 2))
+
+
+class BaseModel(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+
+class Revision(BaseModel):
+    """
+    A representation of a revision in the database.
+
+    Includes a reference to the related Phabricator revision and diff ID if one exists.
+    """
+
+    # revision_id and diff_id map to Phabricator IDs (integers).
+    revision_id = models.IntegerField(null=True, unique=True)
+
+    # diff_id is that of the latest diff on the revision at landing request time. It
+    # does not track all diffs.
+    diff_id = models.IntegerField(null=True)
+
+    # The actual patch.
+    patch_bytes = models.BinaryField(default=b"")
+
+    # Patch metadata, such as author, timestamp, etc...
+    patch_data = models.JSONField(default=dict)
+
+    # A general purpose data field to store arbitrary information about this revision.
+    data = models.JSONField(default=dict)
+
+    def __repr__(self) -> str:
+        """Return a human-readable representation of the instance."""
+        # Add an identifier for the Phabricator revision if it exists.
+        phab_identifier = (
+            f" [D{self.revision_id}-{self.diff_id}]>" if self.revision_id else ""
+        )
+        return f"<{self.__class__.__name__}: {self.id}{phab_identifier}>"
+
+    @property
+    def patch_string(self) -> str:
+        """Return the patch as a UTF-8 encoded string."""
+        return self.patch_bytes.decode("utf-8")
+
+    def set_patch(self, raw_diff: str, patch_data: dict[str, str]):
+        """Given a raw_diff and patch data, build the patch and store it."""
+        self.patch_data = patch_data
+        patch = build_patch_for_revision(raw_diff, **self.patch_data)
+        self.patch_bytes = patch.encode("utf-8")
+
+
+class LandingJob(BaseModel):
+    class Status(models.TextChoices):
```

Comment:
I think in this particular case, it just means that we could name the class `Status` instead of `LandingJobStatus`, and thus refer to it as `LandingJob.Status`. I don't think there's any advantage other than it is largely not needed outside the scope of this model. Happy to move it out of this model as I don't think there's much to gain or lose from this change.